resources/helm/overlays/istio_cni/templates/daemonset-v2.4.yaml

```diff
@@ -67,7 +67,55 @@ spec:
             httpGet:
               path: /readyz
               port: 8000
-          command: ["install-cni"]
+          command:
+            - /bin/sh
+            - -c
+            - |
+              set -xe
+              # Collect host OS information
+              . /host/etc/os-release
+
+              # Detect the major version of the host OS the pod is running on
+              rhelmajor=
+              case "${ID}" in
+                rhcos|scos)
+                  rhelmajor=$(echo $RHEL_VERSION | sed -E 's/([0-9]+)\.{1}[0-9]+(\.[0-9]+)?/\1/')
+                ;;
+                rhel)
+                  rhelmajor=$(echo "${VERSION_ID}" | cut -f 1 -d .)
+                ;;
+                fedora)
+                  if [ "${VARIANT_ID}" == "coreos" ]; then
+                    rhelmajor=8
+                  else
+                    echo "WARNING: Unsupported Fedora variant=${VARIANT_ID}"
+                  fi
+                ;;
+                *)
+                  echo "WARNING: Unsupported OS ID=${ID}"
+                ;;
+              esac
+
+              # Select the appropriate CNI binary to copy based on the OS version
+              sourcebin=
+              case "${rhelmajor}" in
+                8)
+                  sourcebin=/opt/cni/bin/istio-cni-rhel8
+                ;;
+                9)
+                  sourcebin=/opt/cni/bin/istio-cni-rhel9
+                ;;
+                *)
+                  sourcebin=/opt/cni/bin/istio-cni-rhel9
+                  echo "WARNING: RHEL major version rhelmajor=${rhelmajor} is unsupported; falling back to istio-cni binary compiled on latest RHEL version: sourcebin=${sourcebin}"
```

Comment:
This warning is a bit strange on non-RHEL systems, since it says `RHEL major version rhelmajor= is unsupported`. As a user, I would have no idea what this is trying to tell me.

Reply:
Yeah, that makes sense. I could change it to something like...
```
WARNING: The host OS is not supported in FIPS-compliant environments; using istio-cni binary compiled on latest RHEL version: sourcebin=${sourcebin}"
```